### ISSUE TITLE
Allow customization log message for log4j style AbstractLogger

### ISSUE THREAD

**Issue by CSchulz** [mreuvers/typescript-logging#4]
It would be nice if it is possible to use user defined tokens in the log message pattern.

If it should not be part of the library, it would be good to get some documentation how to implement own log formats.

**Comment by mreuvers** [comment]
Hello @CSchulz 

Can you give some example(s) what you are thinking about?

**Comment by CSchulz** [comment]
For example to log a request ID or similar.

In general a possibility to specify another log format or allow to overwrite the createMessage method would be another good way.

**Comment by mreuvers** [comment]
Hello @CSchulz 

Ok, then I assume you are using the log4j variant here (the categorized variant allows customization of the message already). 

The code of log4j is older, but it makes sense to allow it here as well. This will break backwards compatibility for AbstractLogger (those that may currently extend it) - but I need to break it already for the next release anyway, so I better add this in then now as well since we're not on a 1.0.0 release yet.

In short the doLog(..) method will change and will get a full message with all information available, this allows you to log the message/stack in any way you see fit if the standard is not sufficient.

Something like:

`abstract log doLog(msg: Message)`

Where the message interface contains relevant data such as message, stack and level.

I think this would solve your issue as well?

**Comment by CSchulz** [comment]
That sounds good.

It would be good if all information can be accessed atomic to generate a complete own data structure, f.e. remote log appender or just log into console.

Another good point would be the support of params in the template like `log('user %s logged in', username)`.
I am not sure if it is possible to solve with closure as well.

Are there any plans when the next release happens? Any development sources?

**Comment by mreuvers** [comment]
Hello @CSchulz 

For the %s format, I would suggest you to use Typescripts out of the box template scripting support instead. Your line would then look like: ```log(`user ${username} logged in`);```  (Note the use of the back-ticks).

There is currently no hard deadline for a release date, as I am doing this in my spare time next to my normal work. The next planned release is 0.3, and will contain among others console control improvements, and I intend to add this doLog(..) enhancement. I hope to release somewhere this month at least.

**Comment by mreuvers** [comment]
Hello @CSchulz 

I just released 0.3.0, which also contains the enhancement for this issue. Thanks for reporting it! Closing this issue now.
